Address `io/ioutil` depreciation and `d.Set` errcheck in `d/terracurl_request` and `r/terracurl_request`

### Terraform Version

v1.2.8

### Affected Resource(s) or Data Source(s)

- `d/terracurl_request`
- `r/terracurl_request`

### Terraform Configuration Files

Not applicable.

### Debug Output

```console
internal/provider/data_source_curl_request.go:112:7: Error return value of `d.Set` is not checked (errcheck)
        d.Set("response", string(body))
             ^
internal/provider/resource_curl.go:198:7: Error return value of `d.Set` is not checked (errcheck)
        d.Set("response", string(body))

SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details.
```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

From the source directory, run:

```bash
golangci-lint run
```

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- SA1019: https://github.com/golang/go/issues/40025

--- a/internal/provider/data_source_curl_request.go
+++ b/internal/provider/data_source_curl_request.go
@@ -4,7 +4,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
@@ -102,14 +102,16 @@ func dataSourceCurlRequestRead(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	defer resp.Body.Close()
-	body, readErr := ioutil.ReadAll(resp.Body)
+	body, readErr := io.ReadAll(resp.Body)
 	if readErr != nil {
 		return diag.FromErr(readErr)
 	}
 
 	respBody.responseBody = string(body)
 
-	d.Set("response", string(body))
+	if err := d.Set("response", string(body)); err != nil {
+		return diag.FromErr(err)
+	}
 
 	return diags
 }

--- a/internal/provider/resource_curl.go
+++ b/internal/provider/resource_curl.go
@@ -4,11 +4,12 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"io"
+	"net/http"
+
 	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"io/ioutil"
-	"net/http"
 )
 
 func resourceCurl() *schema.Resource {
@@ -175,7 +176,7 @@ func resourceCurlCreate(ctx context.Context, d *schema.ResourceData, meta interf
 	}
 
 	defer resp.Body.Close()
-	body, readErr := ioutil.ReadAll(resp.Body)
+	body, readErr := io.ReadAll(resp.Body)
 	if readErr != nil {
 		return diag.FromErr(readErr)
 	}
@@ -194,7 +195,9 @@ func resourceCurlCreate(ctx context.Context, d *schema.ResourceData, meta interf
 
 	respBody.responseBody = string(body)
 
-	d.Set("response", string(body))
+	if err := d.Set("response", string(body)); err != nil {
+		return diag.FromErr(err)
+	}
 
 	tflog.Trace(ctx, "created a resource")
 
@@ -274,7 +277,7 @@ func resourceCurlDelete(ctx context.Context, d *schema.ResourceData, meta interf
 	}
 
 	defer resp.Body.Close()
-	body, readErr := ioutil.ReadAll(resp.Body)
+	body, readErr := io.ReadAll(resp.Body)
 	if readErr != nil {
 		return diag.FromErr(readErr)
 	}

